doc/build/orm/declarative_mixins.rst

```diff
@@ -510,19 +534,20 @@ function should be invoked **for each class in the hierarchy**, in *almost*
         @declared_attr.cascading
         def id(cls):
             if has_inherited_table(cls):
-                return Column(ForeignKey('person.id'), primary_key=True)
-            else:
```

Comment:
this one is a preference call, and I generally prefer the first way:

```py
if x:
    return a
else:
    return b
```

I find it makes it clear that the two conditions are equally "weighted"

Reply:
Generally it's a good idea to reduce nesting by adding early returns:
```py
def function():
    if condition:
        if another_condition:
            return 42
        else:
            raise ValueError
    else:
        return 0


def function():
    if not condition:
        return 0
    if not another_condition:
        raise ValueError
    return 42
```
If you don't like that i can change it back to if-else

Reply:
well in this case, the max level of nesting doesnt change.    if the "else" condition started garnering a lot more logic, then yes, I'd switch it.  but currently, it looks nicer to me showing these two conditions as equal branch points, since extra logic to increase nesting could just as easily be incurred in the first conditional and not the second, or it could be in both, in which case you'd refactor into two separate methods, there's all kinds of possibilities that nonetheless none are realized.  so don't make that decision early, leave the two branches as equal weight.